packages/process/src/node/raw-process.ts

```diff
@@ -115,28 +115,39 @@ export class RawProcess extends Process {
                 error.code = error.code || 'Unknown error';
                 this.emitOnError(error as ProcessErrorEvent);
             });
-            this.process.on('exit', (exitCode: number, signal: string) => {
+
+            // When no stdio option is passed, it is null by default.
+            this.outputStream = this.process.stdout || new DevNullStream({ autoDestroy: true });
+            this.inputStream = this.process.stdin || new DevNullStream({ autoDestroy: true });
+            this.errorStream = this.process.stderr || new DevNullStream({ autoDestroy: true });
+
+            this.process.on('exit', (exitCode, signal) => {
                 // node's child_process exit sets the unused parameter to null,
                 // but we want it to be undefined instead.
                 this.emitOnExit(
-                    exitCode !== null ? exitCode : undefined,
-                    signal !== null ? signal : undefined,
+                    typeof exitCode === 'number' ? exitCode : undefined,
+                    typeof signal === 'string' ? signal : undefined,
                 );
             });
 
-            this.outputStream = this.process.stdout || new DevNullStream();
-            this.inputStream = this.process.stdin || new DevNullStream();
-            this.errorStream = this.process.stderr || new DevNullStream();
+            this.process.on('close', (exitCode, signal) => {
+                // node's child_process exit sets the unused parameter to null,
+                // but we want it to be undefined instead.
```

Comment:
This was an incorrect assumption, wasn't it? I am not sure if forcing `undefined` is correct
```
root INFO Changed application state from 'ready' to 'closing_window'.
root INFO >>> Storing the layout...
root INFO <<< The layout has been successfully stored.
root ERROR TypeError [ERR_INVALID_ARG_TYPE]: The "err" argument must be of type number. Received type undefined
    at validateNumber (internal/validators.js:130:11)
    at Object.getSystemErrorName (util.js:1435:3)
    at errnoException (internal/errors.js:303:21)
    at ChildProcess.kill (internal/child_process.js:430:26)
    at Object.<anonymous> (C:\Users\kittaakos\dev\my-project\node_modules\@theia\core\lib\node\messaging\ipc-connect
ion-provider.js:55:101)
    at Object.disposable.dispose (C:\Users\kittaakos\dev\my-project\node_modules\@theia\core\lib\common\disposable.j
s:95:13)
    at DisposableCollection.dispose (C:\Users\kittaakos\dev\my-project\node_modules\@theia\core\lib\common\disposabl
e.js:72:40)
    at DisposableCollection.disposable.dispose (C:\Users\kittaakos\dev\my-project\node_modules\@theia\core\lib\commo
n\disposable.js:95:13)
    at DisposableCollection.dispose (C:\Users\kittaakos\dev\my-project\node_modules\@theia\core\lib\common\disposabl
e.js:72:40)
    at FileSystemWatcherServerClient.dispose (C:\Users\kittaakos\dev\my-project\node_modules\@theia\filesystem\lib\n
ode\filesystem-watcher-client.js:55:24)
root INFO >>> Disposing my service...
root INFO <<< Disposed my service.
root ERROR TypeError [ERR_INVALID_ARG_TYPE]: The "err" argument must be of type number. Received type undefined
    at validateNumber (internal/validators.js:130:11)
    at Object.getSystemErrorName (util.js:1435:3)
    at errnoException (internal/errors.js:303:21)
    at ChildProcess.kill (internal/child_process.js:430:26)
    at RawProcess.kill (C:\Users\kittaakos\dev\my-project\node_modules\@theia\process\lib\node\raw-process.js:137:26
)
    at C:\Users\kittaakos\dev\my-project\node_modules\@theia\languages\lib\node\language-server-contribution.js:135:
151
    at Object.dispose (C:\Users\kittaakos\dev\my-project\node_modules\vscode-ws-jsonrpc\lib\server\connection.js:31:
24)
    at Object.clientConnection.onClose [as dispose] (C:\Users\kittaakos\dev\my-project\node_modules\vscode-ws-jsonrp
c\lib\server\connection.js:12:53)
    at DisposableCollection.dispose (C:\Users\kittaakos\dev\my-project\node_modules\vscode-ws-jsonrpc\lib\disposable
.js:15:36)
    at reader.onClose (C:\Users\kittaakos\dev\my-project\node_modules\vscode-ws-jsonrpc\lib\server\connection.js:18:
41)
Done in 363.54s.
```

Happens when I am closing the electron application with a LS running.

Reply:
Discussing on https://github.com/eclipse-theia/theia/issues/6654